.gitignore

```diff
@@ -18,7 +18,8 @@ gradle-app.setting
 *.ipr
 *.iws
 *.uml
-.idea/
+.idea/*
```

Comment:
Why `*` is added here? It should work without the star.

Reply:
Otherwise IntelliJ IDEA doesn't resolve next instruction `!.idea/icon.png` and show similar warning:

<img width="456" alt="Screenshot 2020-04-05 at 15 41 18" src="https://user-images.githubusercontent.com/9942723/78498630-1dd9e980-7754-11ea-8901-6e9019a506f1.png">

Reply:
I agree with @vlsi. If you add the asterisk `*`, commands like `git status -s -u` will show you numerous other files under `.idea` folders across the repo.

Reply:
indeed

Reply:
@sbrannen , I'm afraid you are not quite right.
`git status -s -u` does not show extra files in `.idea` folder.

I've just opened gitignore documentation, and the star is indeed a way to go: https://git-scm.com/docs/gitignore

>/foo/*
>!/foo/bar

Reply:
Hmmm..... strange... on my machine, `git status -s -u` was showing me numerous other files after applying this PR.

Reply:
Cannot reproduce:

```
$ git log -1
commit bc912a7754d25583ee6f11c7e801bdbaf08cf0af (HEAD -> pr2240)
Author: Victor Turansky <victor.turansky@gmail.com>
Date:   Sun Apr 5 15:26:28 2020 +0300

    Add icon for JetBrains Toolbox
$ git status -s -u
?? .editorconfig
?? git_objects_by_size.sh

$ ls -la | grep '.idea'
drwxr-xr-x   16 vladimirsitnikov  staff    512  5 апр 16:04 .idea
```

Reply:
I have `st` as an alias for `status -s -u`, and I'm using Hub that allows me to pull down a PR via `am -3 <URL>`. So with that in mind, here's what happens on my machine.

```shell
[junit5 (master)]$ git pull
Already up to date.
[junit5 (master)]$ git st
[junit5 (master)]$ git am -3 https://github.com/junit-team/junit5/pull/2240
Applying: Add icon for JetBrains Toolbox
[junit5 (master)]$ git st
?? documentation/.idea/.name
?? documentation/.idea/compiler.xml
?? documentation/.idea/copyright/profiles_settings.xml
?? documentation/.idea/gradle.xml
?? documentation/.idea/libraries/Gradle__com_assembla_scala_incubator_graph_core_2_11_1_9_1.xml
?? documentation/.idea/libraries/Gradle__de_schauderhaft_degraph_degraph_check_0_1_3.xml
?? documentation/.idea/libraries/Gradle__de_schauderhaft_degraph_degraph_core_0_1_3.xml
?? documentation/.idea/libraries/Gradle__junit_junit_4_12.xml
?? documentation/.idea/libraries/Gradle__net_sf_jopt_simple_jopt_simple_5_0_2.xml
?? documentation/.idea/libraries/Gradle__org_apache_logging_log4j_log4j_api_2_6_1.xml
?? documentation/.idea/libraries/Gradle__org_apache_logging_log4j_log4j_core_2_6_1.xml
?? documentation/.idea/libraries/Gradle__org_apache_logging_log4j_log4j_jul_2_6_1.xml
?? documentation/.idea/libraries/Gradle__org_apache_maven_surefire_common_java5_2_19_1.xml
?? documentation/.idea/libraries/Gradle__org_apache_maven_surefire_surefire_api_2_19_1.xml
?? documentation/.idea/libraries/Gradle__org_assertj_assertj_core_3_4_1.xml
?? documentation/.idea/libraries/Gradle__org_hamcrest_hamcrest_all_1_3.xml
?? documentation/.idea/libraries/Gradle__org_hamcrest_hamcrest_core_1_3.xml
?? documentation/.idea/libraries/Gradle__org_mockito_mockito_core_1_10_19.xml
?? documentation/.idea/libraries/Gradle__org_objenesis_objenesis_2_1.xml
?? documentation/.idea/libraries/Gradle__org_opentest4j_opentest4j_1_0_0_M1.xml
?? documentation/.idea/libraries/Gradle__org_ow2_asm_asm_5_0_3.xml
?? documentation/.idea/libraries/Gradle__org_scala_lang_modules_scala_parser_combinators_2_11_1_0_3.xml
?? documentation/.idea/libraries/Gradle__org_scala_lang_modules_scala_xml_2_11_1_0_3.xml
?? documentation/.idea/libraries/Gradle__org_scala_lang_scala_library_2_11_6.xml
?? documentation/.idea/libraries/Gradle__org_scala_lang_scala_reflect_2_11_4.xml
?? documentation/.idea/libraries/Gradle__org_scalatest_scalatest_2_11_2_2_4.xml
?? documentation/.idea/libraries/Gradle__org_spockframework_spock_core_1_0_groovy_2_4.xml
?? documentation/.idea/misc.xml
?? documentation/.idea/modules.xml
?? documentation/.idea/vcs.xml
?? documentation/.idea/workspace.xml
```

🤔

Reply:
It is strange that you have multiple `.idea` directories. Typically the one in the root is enough.

Reply:
@sbrannen Is `documentation` a separate project?

Reply:
TBH, I don't actively use IDEA, but I obviously imported some projects into IDEA at some point, and that's simply what's in the file system.

In any case, isn't is possible to craft the patterns so that they apply to `.idea` directories in subdirectories as well?

Reply:
> I obviously imported some projects into IDEA at some point, and that's simply what's in the file system.

In this case IDEA will download Gradle for you  (possibly incompatible with your root project) and root project settings will not be applied to your subproject.

For example `documentation` isn't separate project.

> In any case, isn't is possible to craft the patterns so that they apply to .idea directories in subdirectories as well?

I guess not

Reply:
Technically speaking, the following would ignore `.idea` everywhere, except it would keep `/.idea/icon.png`.

```
**/.idea/*
!/.idea/icon.png
```

Of course, it is up to you, however, I'm inclined that `.idea` from sub-folders should be removed.

Reply:
I'd personally rather go with the more explicit (and IMO correct) patterns.

Whether or not somebody has `.idea` folders in subfolders is up to that individual. Maybe it's an accident, but maybe it's intentional.

@junit-team/junit-lambda Thoughts?

Reply:
> I'd personally rather go with the more explicit (and IMO correct) patterns.

@sbrannen Done  :)